### DISCUSSION TITLE
How to subscribe product variant stock updates with a webhook ?

### DISCUSSION THREAD

**Discussion by keremdokumaci** [saleor/saleor#11622] [Q&A]
Hello,

I want to subscribe product variant stock updates with an async webhook. Afaik, there is no async webhook which is directly related with product variant stocks.

I tried to subscribe **PRODUCT_VARIANT_UPDATED** webhook with given mutation.
`
mutation webhookCreate {
  webhookCreate(
    input: {
      app: "<app-id>"
      name: "product_variant_updated_subscription"
      targetUrl: "<target>"
      asyncEvents: [PRODUCT_VARIANT_UPDATED]
      query: "subscription{event{...on ProductVariantUpdated{productVariant{id updatedAt quantityAvailable sku}}}}"
    }
  ) {
    errors {
      field
      message
    }
    webhook {
      id
    }
  }
}
`

It works when I update product variant inventory quantity by using dashboard. However, I have some flow that automatically updates stocks with given mutation below.

`
mutation postProductVariantStockUpdate($warehouse: ID!, $quantity:Int!, $variant_id:ID!) {
      productVariantStocksUpdate(
        stocks: {
          warehouse: $warehouse
          quantity: $quantity
        }
        variantId: $variant_id
      ) {
        productVariant {
          id
          stocks {
            warehouse {
              id
            }
            quantity
          }
        }
        errors {
          field
          message
        }
      }
    }
`
It doesn't trigger the **PRODUCT_VARIANT_UPDATED** webhook.

Where is my mistake ? Do you have any idea ?

**Comment by peelar** [comment]
Hello @keremdokumaci 👋 

Thanks for pointing it out, it is a bug. I will create an issue for it, reference your post and let you know once we have fixed it.

**Reply by keremdokumaci** [reply]
Thanks !

**Reply by timuric** [reply]
You can track this issue here https://github.com/saleor/saleor/issues/11630

**Reply by keremdokumaci** [reply]
👍
